docs/_newsfragments/452.breakingchange.rst

```diff
@@ -0,0 +1,6 @@
+The class :class:`~.falcon.http_error.OptionalRepresentation` and the attribute
+:attr:`~.falcon.HTTPError.has_representation` were removed. The default error
+serializer now generates a representation for every error type that derives from
+:class:`falcon.HTTPError`. In addition, Falcon now ensures that any previously
+set response body is cleared before handling any subclass of
+:class:`falcon.HTTPStatus` or any other raised exception.
```

Comment:
Not sure the latter part of the statement is universally true.

Consider, for instance, the following app:
```python
import falcon


class Resource:

    def on_get(self, req, resp):
        resp.media = {'message': 'Hello, World!'}
        resp.set_header('X-Value', str(0/0))


def handle_zero_division(req, resp, ex, params):
    print('resp.media ==', resp.media)
    print('This went wrong:', ex)


app = falcon.App()
app.add_error_handler(ZeroDivisionError, handle_zero_division)
app.add_route('/test', Resource())
```
```
resp.media == {'message': 'Hello, World!'}
This went wrong: division by zero
```
I.e., it seems that when handling _any other_ raised exception, the response body is not necessarily cleared.

Reply:
Nice catch. I'll fix that, since my intention was to do as in the note :)